src/dual.jl

```diff
@@ -1,68 +1,67 @@
-immutable Dual{T<:Real} <: Number
-    re::T
-    du::T
+typealias ReComp Union{Real,Complex}
+
+immutable Dual{T<:ReComp} <: Number
+    value::T
+    dual::T
 end
-Dual(x::Real, y::Real) = Dual(promote(x,y)...)
-Dual(x::Real) = Dual(x, zero(x))
+Dual{T<:ReComp}(x::T, y::T) = Dual{T}(x, y)
+
+Dual{S<:ReComp,T<:ReComp}(x::S, y::T) = Dual(promote(x,y)...)
+Dual{T<:ReComp}(x::T) = Dual(x, zero(x))
 
-const du = Dual(false, true)
+const ɛ = Dual(false, true)
 
 typealias Dual128 Dual{Float64}
 typealias Dual64 Dual{Float32}
 typealias DualPair Dual
```

Comment:
I know this isn't exactly in the scope of this PR, but is there any reason to keep around `DualPair` (I'm not sure why it exists in the first place)?